### PR TITLE
Merge feature/rollback-on-faulty-restart into master

### LINKED ISSUE #3: Check custom.conf before restarting dnsmasq
Apply a check to the configuration file by firing the following command:
```
dnsmasq --test
```

If the test was successful, restart dnsmasq. If the test failed revert the custom.conf file and return a `409 Conflict` error.

### PR DESCRIPTION
**Pull request by mbuelowdev**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -102,6 +104,7 @@ async fn action_delete(item: web::Json<EntryRequestDto>) -> HttpResponse {
 
     // Load entries from custom.conf
     let mut custom_entries = load_custom_dnsmasq_entries_from_file();
+    let custom_entries_original = custom_entries.clone();
```
Comment (Orga-Space) [inline src/main.rs:109]: why not in own function

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "api-dnsmasq-dynconf"
-version = "1.0.0"
+version = "1.1.0"
```
Comment (Orga-Space) [inline Cargo.toml:3]: Is this the right version increase[?](https://semver.org/)

File: src/main.rs (hunk 1 above)
Comment (mbuelowdev) [reply src/main.rs:109]: It is not possible to abstract this code any further. At least not with my Rust knowledge as of today.

File: Cargo.toml (hunk 2 above)
Comment (mbuelowdev) [reply Cargo.toml:3]: This is indeed incorrect versioning. I forgot to return a fitting http response code to justify a minor version increase.

**Comment by mbuelowdev** [comment]
The `/add` and `/delete` endpoints will now return a `400 Bad Request` in case dnsmasq is not able to restart. This should justify a minor version increase (@Orga-Space).

File: src/main.rs (hunk 1 above)
Comment (Orga-Space) [reply src/main.rs:109]: ok

**Comment by Orga-Space** [comment]
@42Pollux, ok. Could 409 be better?

**Review by Orga-Space — APPROVED** [review]

**Pull request merged**